Kudu.Contracts/Functions/FunctionSecretsJsonOps.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+
+namespace Kudu.Core.Functions
+{
+    public class FunctionSecretsJsonOps : IKeyJsonOps<FunctionSecrets>
+    {
+        public int RequireKeyCount()
+        {
+            return 1;
+        }
+
+        // have the schema related info enclosed in this class
+        public string GenerateKeyUglyJson(Tuple<string, string>[] keyPair, string functionRt, out string unencryptedKey)
+        {
+            unencryptedKey = keyPair[0].Item1;
+            if (string.CompareOrdinal(functionRt, "~0.7") < 0)
+            {
+                return $"{{\"key\":\"{unencryptedKey}\"}}";
+            }
+            else
+            {
+                return $"{{\"keys\":[{{\"name\":\"default\",\"value\":\"{keyPair[0].Item2}\",\"encrypted\": true }}]}}";
+            }
+        }
+
+        public string GetKeyInString(string json, out bool isEncrypted)
+        {
+            try
+            {
+                JObject hostJson = JObject.Parse(json);
+                if (hostJson["key"]?.Type == JTokenType.String)
+                {
+                    isEncrypted = false;
```

Comment:
there are two assignment statement, but it is only set once (if old format, we default to not encrypted, else we read the `encrypted` field in the new key format)